codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsProtocolUtils.java

```diff
@@ -91,6 +85,20 @@ static void generateHttpProtocolTests(GenerationContext context) {
                         .build()
         ));
 
+        // TODO can this check be replaced with a lookup into the runtime plugins?
+        if (IdempotencyTokenMiddlewareGenerator.hasOperationsWithIdempotencyToken(context.getModel(), context.getService())) {
+            configValues.add(
+                    HttpProtocolUnitTestGenerator.ConfigValue.builder()
+                            .name(IdempotencyTokenMiddlewareGenerator.IDEMPOTENCY_CONFIG_NAME)
```

Comment:
To resolve this issue i think @skotambkar RESTXML pr moved the idempotency token config value to always be set for all AWS protocol test client option values.

Reply:
The json rpc protocol tests don't have any idempotency token tests in them, so the config option doesn't have it.